extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyRowFilterAndDataMasking.scala

```diff
@@ -30,17 +34,45 @@ import org.apache.kyuubi.plugin.spark.authz.util.RowFilterAndDataMaskingMarker
 class RuleApplyRowFilterAndDataMasking(spark: SparkSession) extends Rule[LogicalPlan] {
 
   override def apply(plan: LogicalPlan): LogicalPlan = {
-    plan transformUp {
-      case hiveTableRelation if hasResolvedHiveTable(hiveTableRelation) =>
-        val table = getHiveTable(hiveTableRelation)
-        applyFilterAndMasking(hiveTableRelation, table, spark)
-      case logicalRelation if hasResolvedDatasourceTable(logicalRelation) =>
-        val table = getDatasourceTable(logicalRelation)
-        if (table.isEmpty) {
-          logicalRelation
+    // Wrap HiveTableRelation/LogicalRelation with RowFilterAndDataMaskingMarker if it is not
```

Comment:
we can change this func to a simplier one

```scala
override def apply(plan: LogicalPlan): LogicalPlan = {
    plan mapChildren {
      case p: RowFilterAndDataMaskingMarker => p
      case hiveTableRelation if hasResolvedHiveTable(hiveTableRelation) =>
        val table = getHiveTable(hiveTableRelation)
        applyFilterAndMasking(hiveTableRelation, table, spark)
      case logicalRelation if hasResolvedDatasourceTable(logicalRelation) =>
        val table = getDatasourceTable(logicalRelation)
        if (table.isEmpty) {
          logicalRelation
        } else {
          applyFilterAndMasking(logicalRelation, table.get, spark)
        }
      case other => apply(other)
    }
  }
```

Reply:
Looks nice!

Reply:
cool！